PhysicsTools/PatAlgos/plugins/BaseMVAValueMapProducer.h

```diff
@@ -109,21 +158,48 @@ BaseMVAValueMapProducer<T>::produce(edm::Event& iEvent, const edm::EventSetup& i
   edm::Handle<edm::View<T>> src;
   iEvent.getByToken(src_, src);
   readAdditionalCollections(iEvent,iSetup);
-  
-  std::vector<float> mvaOut;
-  mvaOut.reserve(src->size());
+  std::vector<std::vector<float>> mvaOut((tmva_)?1:output_names_.size());
+  for( auto & v : mvaOut) v.reserve(src->size());
+
   for(auto const & o: *src) {
 	for(auto const & p : funcs_ ){
-		values_[positions_[p.first]]=p.second(o);
+        setValue(p.first,p.second(o));
 	}
         fillAdditionalVariables(o);
-	mvaOut.push_back(isClassifier_ ? reader_.EvaluateMVA(name_) : reader_.EvaluateRegression(name_)[0]);
+    if(tmva_){
+        mvaOut[0].push_back(isClassifier_ ? reader_->EvaluateMVA(name_) : reader_->EvaluateRegression(name_)[0]);
+    }
+    if(tf_){
+        //currently support only one input sensor to reuse the TMVA like config 
+        tensorflow::TensorShape input_size   {1,(long long int)positions_.size()} ;
+        tensorflow::NamedTensorList input_tensors;
+        input_tensors.resize(1); 
+        input_tensors[0] = tensorflow::NamedTensor(inputTensorName_, tensorflow::Tensor(tensorflow::DT_FLOAT, input_size));
+        for(size_t j =0; j < values_.size();j++) {
+           input_tensors[0].second.matrix<float>()(0,j) = values_[j];
+        }
+       std::vector<tensorflow::Tensor> outputs;
+       std::vector<std::string> names; names.push_back(outputTensorName_);
+       tensorflow::run(session_, input_tensors, names, &outputs);
+//       for(size_t k=0;k<output_names_.size();k++) mvaOut[k].push_back(outputs.at(0).matrix<float>()(0, k));
```

Comment:
is this commented out code needed?
Please remove or add comments inline in the code why the commented out block is relevant

Reply:
I guess it can be removed